src/realm/util/file.cpp

```diff
@@ -198,6 +208,18 @@ void remove_dir_recursive(const std::string& path)
 }
 
 
+bool try_remove_dir_recursive(const std::string& path)
```

Comment:
It is kind of annoying that `try_remove_dir_recursive` is implemented in terms of `remove_dir_recursive`, whereas `remove_dir` is implemented in terms of `try_remove_dir`. Can that be done differently?